src/pages/MerchantGiftCards/MerchantGiftCards.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import {
+  getMerchantGiftCards,
+  getSeller,
+} from '../../utilities/api/interactionManager';
+import Loader from '../../components/Loader/Loader';
+import styles from './styles.module.scss';
+import ErrorPage from '../../components/404Page';
+const FilterableTable = require('react-filterable-table');
+
+const MerchantGiftCards = () => {
+  const params = useHistory();
+  let urlParams = params.location.pathname.match(/\/[^/]+/g) as string[];
+  urlParams = urlParams.map((param) => param.replace('/', ''));
+  const metadata = {
+    sellerId: urlParams[0],
+    secretId: urlParams[2],
+  };
+
+  const [error, setError] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [giftCards, setGiftCards] = useState<any | null>();
+  const [seller, setSeller] = useState<any | null>();
+
+  const renderAmount = (props) => {
+    return Intl.NumberFormat('en-US', {
+      style: 'currency',
+      currency: 'USD',
+      maximumSignificantDigits: 4,
+    }).format(props.value / 100);
+  };
+
+  const renderDate = (props) => {
+    return new Date(props.value).toISOString().substring(0, 10);
+  };
+
+  const fields = [
+    {
+      name: 'seller_gift_card_id',
+      displayName: 'Voucher Code \n 礼品券号码',
+      inputFilterable: true,
+      sortable: true,
+    },
+    {
+      name: 'email',
+      displayName: 'Email \n 电子邮件',
+      inputFilterable: true,
+      sortable: true,
+    },
+    {
+      name: 'value',
+      displayName: 'Original Amount \n 购买金额',
+      inputFilterable: true,
+      sortable: true,
+      render: renderAmount,
+    },
+    {
+      name: 'created_at',
+      displayName: 'Date Purchased \n 购买日期',
+      inputFilterable: true,
+      sortable: true,
+      render: renderDate,
+    },
+    /* {
+      name: 'expiration',
+      displayName: 'Expiration',
+      inputFilterable: true,
+      sortable: true,
+    },
+    Uncomment if we want expiration */
+  ];
+
+  const fetchData = async () => {
+    try {
+      const giftCardData = (
+        await getMerchantGiftCards(metadata.sellerId, metadata.secretId)
+      ).data;
+      const sellerData = (await getSeller(metadata.sellerId)).data;
+
+      setGiftCards(giftCardData);
+      setSeller(sellerData);
+    } catch {
+      setError(true);
+    }
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    setLoading(true);
```

Comment:
this call shouldnt be necessary if loading is already true when the component is mounted onto the DOM

Reply:
Ah good catch, I changed loading to be true by default and didn't remove this 👍